cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -15,10 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#ifndef ARROW_UTIL_BYTE_STREAM_SPLIT_H
-#define ARROW_UTIL_BYTE_STREAM_SPLIT_H
+#pragma once
 
-#include "arrow/util/sse_util.h"
+#include "arrow/util/byte_stream_split_simd_internal.h"
```

Comment:
I don't think there's any reason for a separate header. This one is already specialized enough.